packages/client/hmi-client/src/model-representation/petrinet/nested-petrinet-renderer.ts

```diff
@@ -0,0 +1,136 @@
+import { select } from 'd3';
+import { D3SelectionINode, IGraph } from '@graph-scaffolder/types';
+import { Options } from '@graph-scaffolder/core/renderer';
+
+import { NodeType, PetrinetRenderer } from '@/model-representation/petrinet/petrinet-renderer';
+import { NodeData } from '@/model-representation/petrinet/petrinet-service';
+
+type AsyncFunction<A, O> = (args: A) => Promise<O>;
+type AsyncLayoutFunction<V, E> = AsyncFunction<IGraph<V, E>, IGraph<V, E>>;
+type LayoutFunction<V, E> = (args: IGraph<V, E>) => IGraph<V, E>;
+
+export interface NestedPetrinetOptions {
+	el?: HTMLDivElement | null;
+	nestedMap?: { [baseNodeId: string]: string[] };
+	runLayout: AsyncLayoutFunction<any, any> | LayoutFunction<any, any>;
+
+	useZoom?: boolean;
+	zoomRange?: [number, number];
+	useStableLayout?: boolean;
+
+	// Attempt to use the same set of zoom parameters across layout changes
+	useStableZoomPan?: boolean;
+
+	// This is getting around algorithms that do not provide stand-alone routing capabilities, in
+	// which case we can internally route using A-star
+	useAStarRouting?: boolean;
+
+	// Whether to show grid
+	useGrid?: boolean;
+
+	// The css class for custom dragging event
+	dragSelector?: string;
+}
```

Comment:
Slightly prefer if we can just extend Options interface .... or maybe something like a setNestedMap function.

We can export the Options interface in graph-scaffolder if that makes things easier